detox/android/detox/src/test/java/com/wix/detox/espresso/scroll/SwiperHelperSpec.kt

```diff
@@ -0,0 +1,131 @@
+package com.wix.detox.espresso.scroll
+
+import android.content.Context
+import android.content.res.Resources
+import android.util.DisplayMetrics
+import android.view.View
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.*
+import com.nhaarman.mockitokotlin2.mock
+import com.wix.detox.espresso.common.annot.*
+import org.mockito.Mockito.`when`
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import kotlin.test.assertEquals
+
+object SwiperHelperSpec: Spek({
+    describe("SwipeHelper") {
+        val view = mock<View>()
+
+        var viewX = 1000
+        var viewY = 2000
+        var viewWidth = 2000
+        var viewHeight = 1000
+        var screenWidth = 4000
+        var screenHeight = 5000
+
+        fun viewLeft() = viewX + viewWidth * 0f;
+        fun viewCenter() = viewX + viewWidth * 0.5f;
+        fun viewRight() = viewX + viewWidth * 1f;
+        fun viewTop() = viewY + viewHeight * 0f;
+        fun viewMiddle() = viewY + viewHeight * 0.5f;
+        fun viewBottom() = viewY + viewHeight * 1f;
+        fun viewFuzzH() = viewWidth * SwipeHelper.edgeFuzzFactor;
+        fun viewFuzzV() = viewHeight * SwipeHelper.edgeFuzzFactor;
+
+        fun screenTop() = 0f;
+        fun screenLeft() = 0f;
+        fun screenBottom() = screenTop() + screenHeight
+        fun screenRight() = screenLeft() + screenWidth
+
+        beforeGroup {
+            val mockDisplayMetrics = mock<DisplayMetrics>()
+            mockDisplayMetrics.widthPixels = screenWidth
+            mockDisplayMetrics.heightPixels = screenHeight
+
+            val mockResources = mock<Resources>()
+            `when`(mockResources.displayMetrics).then { mockDisplayMetrics }
```

Comment:
Sorry for nitpicking, but please switch to `whenever` instead of \`when\`. When comes from mockito's core rather than the mockito-kotlin wrapper, and is not as idiomatic.